### ISSUE TITLE
BO - Orders - Problems on product price changing in database

### ISSUE THREAD

**Issue by ytilotti** [PrestaShop/PrestaShop#24529]
#### Describe the bug

If I change the price on a product in an order via the administration, the values of `product_price` and `original_product_price` are different or same between the method used for update.

#### Expected behavior

The same behavior on differents methods

##### Create a first order:

1. Go to admin > order settings > order settings
2. Click on **add new order**
3. Choose a customer
4. Search a product and add to cart. Mug The Adventure Begins for example at 11.90
5. Change the price to 8.00
6. Choose the payment method & status and create the order
7. In your database check the `product_price`, `original_product_price` and `unit_price_tax_excl` in the `order_detail` table.

product_price = 11.90
original_product_price = 11.90
unit_price_tax_excl = 8.00

##### Create a second order:

Do the same steps but don't change the price at step 5. Create the order.

1. On the order view edit the price of the product at 8.00 
2. Save the line and confirm
3. In your database check the `product_price`, `original_product_price` and `unit_price_tax_excl` in the `order_detail` table.

product_price = 8.00
original_product_price = 11.90
unit_price_tax_excl = 8.00

##### Conclusion:

There is a difference between the two methods, I don't know which method is in error.

#### Additional information

* PrestaShop version: 1.7.7.4
* PHP version: 7.2.34
* MariaDB: 10.5.9

**Comment by hibatallahAouadni** [comment]
Hello @ytilotti 

> 1. Go to admin > order settings > order settings

I know it's not about the issue, but I just need to verify this wording. We do not have `Order settings > Order settings`, we have `Orders > Orders`, check the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/118512758-60ca0700-b733-11eb-8674-a430a939cdd4.png)

Now, about your issue. I reproduce it with PS1.7.7.0, PS1.7.7.4 and develop, see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/118517859-e18b0200-b737-11eb-902a-1d7a0758c9b2.png)

But I don't manage to reproduce it with PS1.7.6.9, so it's a regression:

![image](https://user-images.githubusercontent.com/23308427/118517957-f9fb1c80-b737-11eb-8477-c40548e26404.png)

I’ll add this to the debug backlog so that it’s fixed.

Please be aware that due to the high number of reported issues, some bugs might take a very long time to be resolved. If this issue is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you. 

If you manage to get the problem fixed, then please contribute the fix back to the project as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thank you

**Comment by jolelievre** [comment]
Hi @ytilotti 

after some research I realized we were wrong in expecting `product_price` to be equal to `original_product_price` When you look at this code:
https://github.com/PrestaShop/PrestaShop/blob/948ecce8461176c507512c977633731bc3550a24/classes/order/Order.php#L607

you can see that it's actually always computed to be equal to `unit_price_tax_excl`

So in the proposed scenario there was indeed an error, but it's not when the price is edited in the order, rather it was on the initial creation, following your example if you edit the price in the "Create an order" page then `product_price` should be `8.0` on the first creation and `original_product_price` should still be `11.90`

@hibatallahAouadni please take this into account when performing the QA:

tldr; `product_price` should ALWAYS be equal to `unit_price_tax_excl` and `original_product_price` should never change event when we edit the product price (in cart or order)

**Comment by Progi1984** [comment]
Fixed by @jolelievre in #24991
